src/mesh/NodeDB.cpp

```diff
@@ -982,9 +984,18 @@ void NodeDB::loadFromDisk()
 #ifdef FSCom
     spiLock->lock();
     if (FSCom.exists(legacyPrefFileName)) {
+        spiLock->unlock();
+        LOG_WARN("Legacy prefs version found, factory resetting");
+        if (loadProto(configFileName, meshtastic_LocalConfig_size, sizeof(meshtastic_LocalConfig), &meshtastic_LocalConfig_msg,
+                      &config) == LoadFileResult::LOAD_SUCCESS &&
+            config.has_security && config.security.private_key.size > 0) {
+            LOG_DEBUG("Saving backup of security config and keys");
+            backupSecurity = config.security;
+        }
+        spiLock->lock();
         rmDir("/prefs");
+        spiLock->unlock();
     }
```

Comment:
If FSCom.exists(legacyPrefFileName) and that branch is skipped --> spi won't be unlocked?

Reply:
Good call. I just ran into that condition locally here on second boot 😆 
Good news is that config restore seemed to work great